### ISSUE TITLE
U-2-Net model different with your paper description

### ISSUE THREAD

**Issue by laoyoutiaotiao** [NathanUA/U-2-Net#52]
[https://github.com/NathanUA/U-2-Net/blob/master/model/u2net.py](url)

        d1 = self.side1(hx1d)
        
        d2 = self.side2(hx2d)
        d2 = _upsample_like(d2,d1)

        d3 = self.side3(hx3d)
        d3 = _upsample_like(d3,d1)

        d4 = self.side4(hx4d)
        d4 = _upsample_like(d4,d1)

        d5 = self.side5(hx5d)
        d5 = _upsample_like(d5,d1)

        d6 = self.side6(hx6)
        d6 = _upsample_like(d6,d1)

        d0 = self.outconv(torch.cat((d1,d2,d3,d4,d5,d6),1))

        return F.sigmoid(d0), F.sigmoid(d1), F.sigmoid(d2), F.sigmoid(d3), F.sigmoid(d4), F.sigmoid(d5), F.sigmoid(d6)
generates six side output saliency probability maps from stages En6, De5, De4, De3, De2 and De1 by a 3x3 convolution layer, without sigmoid function.
 But in your paper  description，generates six side output saliency probability maps from stages En6, De5, De4, De3, De2 and De1 by a 3x3 convolution layer and a sigmoid function. 

Why the difference?

**Comment by bluesky314** [comment]
There seems to be sigmoids here:
" return F.sigmoid(d0), F.sigmoid(d1), F.sigmoid(d2), F.sigmoid(d3), F.sigmoid(d4), F.sigmoid(d5), F.sigmoid(d6)"

**Comment by laoyoutiaotiao** [comment]
in the last convolution layer 'd0 = self.outconv(torch.cat((d1,d2,d3,d4,d5,d6),1))', d1~d6 just from stages En6~De1 by a 3x3 convolution layer , without sigmoid function. Is that understanding？

**Comment by NathanUA** [comment]
Thanks for you insightful comments. Yes, here the code is a bit different from the description in the paper. Because drawing both logits and probability maps takes more space in the figure of the paper. For the simplicity, we just draw the probability maps (of d1-d6) there and then gave inaccurate descriptions based on the figure. We are sorry about that. Please follow the released code for usage.
